### PR TITLE
Upgrade polarion ids

### PR DESCRIPTION
**Pull request by clacroix12**
Fixes: #2803

### CODE REVIEW

**Review by fbalak — APPROVED** [review]

**Comment by clacroix12** [comment]
Currently the upgrade_id isn't being utilized because `config.UPGRADE.get('upgrade'):` is evaluating to `False`. This is because we aren't passing a configuration when we execute the upgrade that sets that config value to `True`.

@petr-balogh is there a reason we aren't passing a config file that sets `config.UPGRADE['upgrade']` to `True` when we run the upgrade stage? I'm assuming we just don't have anything that is hooking onto that config value yet so we haven't needed to pass that config. I'm going to raise an MR that adds that config to the upgrade stage so we know from the config that we are executing an upgrade. If we aren't adding that value for a particular reason I will have to change how we determine we are running an upgrade.

**Comment by clacroix12** [comment]
Verified from the upgrade xml that the polarion ID is being added as a custom property: http://magna002.ceph.redhat.com/ocsci-jenkins/openshift-clusters/jnk-pr2805-b2757/jnk-pr2805-b2757_20200831T225536/logs/upgrade_test_results_1598914396.xml

File: ocs_ci/utility/reporting.py [hunk 1]
```diff
@@ -1,20 +1,28 @@
...
     polarion_config = config.REPORTING.get('polarion')
     if polarion_config:
-        if config.UPGRADE.get('upgrade'):
-            if config.DEPLOYMENT.get('subscription_plan_approval') == 'Manual':
-                return polarion_config.get('upgrade_manual_id')
-            else:
-                return polarion_config.get('upgrade_auto_id')
+        if upgrade:
+            upgrade_id = polarion_config.get('upgrade_id')
+            log.info('polarion upgrade_id: %s', upgrade_id)
```
Comment (AviadP) [inline ocs_ci/utility/reporting.py:23]:
```suggestion
            log.info(f'polarion upgrade_id: {upgrade_id}')
```

File: ocs_ci/utility/reporting.py [hunk 2]
```diff
@@ -1,20 +1,28 @@
...
+            return upgrade_id
         else:
-            return polarion_config.get('deployment_id')
+            deployment_id = polarion_config.get('deployment_id')
+            log.info('polarion deployment_id: %s', deployment_id)
```
Comment (AviadP) [inline ocs_ci/utility/reporting.py:27]:
```suggestion
            log.info(f'polarion deployment_id: {deployment_id}')
```

File: ocs_ci/utility/reporting.py (hunk 1 above)
Comment (fbalak) [reply ocs_ci/utility/reporting.py:23]: Those log messages are in line with our code guidelines https://github.com/red-hat-storage/ocs-ci/blob/a94361d4f30b3a1901ce376f0e85048c26707e78/docs/coding_guidelines.md for Logging.

Comment (clacroix12) [reply ocs_ci/utility/reporting.py:23]: Correct, when it comes to logging we have decided to let the logger handle the message formatting.

**Review by vasukulkarni — APPROVED** [review]

**Review by mbukatov — APPROVED** [review]

**Review by petr-balogh — APPROVED** [review]

**Review by ebenahar — APPROVED** [review]

**Comment by petr-balogh** [comment]
@clacroix12 based on our latest conversation I guess this we didn't want to be merged right?

**Pull request merged**